### ISSUE TITLE
PSBT with seedsigner fails

### ISSUE THREAD

**Issue by pony-montana** [akumaigorodski/wallet#165]
The problem accour in the final step of PSBT, when the seedsigner generates the qr containing the signed txo. SBW isnt able to recognize the qr and shows an error. It shows a load screen when scanning qr that end at 99% with the fatal error.

The seesigner signin seems to work (no error reported, no unexpected behaviour and the final qr is correctly generated).

**Comment by pony-montana** [comment]
https://github.com/nbd-wtf/obw/issues/47 here I presented the issue to obw, a sbw fork. SBW behaviour was identical to OBW, so in the discussion you can find some useful informations.

**Comment by pony-montana** [comment]
now I cant import hardware wallet, parsing from qr generated by seedsigner fails (worked well in pre 2.5 release). 
The error says "next on empty iterator". I don't know if this happens with other hardware wallets (I think yes), I add this comment here and I don't create a new issue because I can test only seedsigner.

**Comment by akumaigorodski** [comment]
Waiting for hardware parts to arrive so I can do some debugging, fix is probably a few weeks away.

**Comment by pony-montana** [comment]
This is the most bullish news I ever heard in the bitcoin space 😍

**Comment by akumaigorodski** [comment]
Unfortunately I could not get hardware to work (blank screen on boot, tried a few different pi models and sd cards). I don't have much free time to keep trying so shelving this for now. If anyone can send me a definitely working hardware setup though then I'll look into it.

**Comment by pony-montana** [comment]
if you tried different pi models and different sd card there are two possible cause for it not working:
1) the screen is broken.
2) there are some trivial errors in the configuration that can be easily catched.

Building seedsigner the first time is not easy, so there are possibilities that we are in the case 2.
In particular, I encountered some common problems building it:

1) pins and screen are a pain. The screen need to be at a "perfect distance", somethimes it doesnt work if the pins are "fully pulled". So, you can try boot the seedsigner with the screen pulled at "different heigths".
I think there are a lot of possibilities that this is the problem.

2) another common problem if you soldered the pin on the pi yourself, is that the board could be dirty. Try to clean it with alcool.

If these solutions doesnt work, you can reach me out, we can help to debug the problem.

**Comment by akumaigorodski** [comment]
I did not yet solder the screen, just connected it such that it clearly gets powered (the screen lights up overall) so I'm not sure, could be screen, are there any ways to diagnose this?

**Comment by pony-montana** [comment]
on boot the camera is initialized and a camera led will lights up. If it doesnt lights up, probably the boot is not happening correctly, and the problem is not the screen (could be the sd card not flashed properly or something else). If it lights up, there are a lot of probability that its a problem with the screen.

**Comment by pony-montana** [comment]
you mean that you just placed the pins on the board without soldering them? If I understood correctly, this is definitevely the problem. The pin connections from board to screen works only after a perfect soldering, they are very fragile. I had a lot of headhache with the screen and with some pins that was "not perfectly soldered" the first time I built it.
If soldering is a problem, you can try solderless pins. They also are a bit problematic (require to push with a hammer the pins in the board) but with a lot of careful they can do the job without requiring special equipment or skills.

**Comment by akumaigorodski** [comment]
Soldering is not a problem, I'll try that.

**Comment by Resikoakun** [comment]
#172

**Comment by akumaigorodski** [comment]
@Resikoakun if you're wondering what has happened to that one, it's been postponed indefinitely due to the lack of time.

**Comment by akumaigorodski** [comment]
Harware wallets support will be removed in future release.
